src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -530,5 +533,98 @@ protected override void OnMarked(NodeFactory context)
         {
             //Debug.Assert(_type.IsTypeDefinition || !_type.HasSameTypeDefinition(context.ArrayOfTClass), "Asking for Array<T> EEType");
         }
+
+        /// <summary>
+        /// Validates that it will be possible to create an EEType for '<paramref name="type"/>'.
+        /// </summary>
+        public static void CheckCanGenerateEEType(TypeDesc type)
+        {
+            // TODO: a set of rules for "constructed" types: e.g. ban creating a constructed EEType for '<Module>' type,
+            //       pointer types, function pointer types, byrefs, generic definitions, etc.
+
+            // It must be possible to create an EEType for the base type of this type
+            TypeDesc baseType = type.BaseType;
+            if (baseType != null)
+            {
+                CheckCanGenerateEEType(baseType);
+            }
+
+            // Don't validate __System.Canon since it is not present in the output image
+            if (type.IsCanonicalSubtype(CanonicalFormKind.Specific))
```

Comment:
This was needed when the `IsGenericDefinition` check came after the base type check.  Removed.